### PR TITLE
[#B-08892] Redis Cluster adapter implementation

### PR DESCRIPTION
**Pull request by tsompos**

### CODE REVIEW

File: library/Zend/Cache/Backend/Rediscluster.php [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+		],
+		'timeout' => self::DEFAULT_TIMEOUT,
+		'read_timeout' => self::DEFAULT_READ_TIMEOUT,
+		'key_prefix' => ''
+	];
+
+	/**
+	 * Redis object
+	 *
+	 * @var mixed redis object
```
Comment (vaionicle) [inline library/Zend/Cache/Backend/Rediscluster.php]:
```
@var RedisCluster|null
```

File: library/Zend/Cache/Backend/Rediscluster.php [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+	 *
+	 * In case of multiple tags, a logical AND is made between tags
+	 *
+	 * @param array $tags array of tags
+	 * @return array array of matching cache ids (string)
+	 */
+	public function getIdsMatchingTags($tags = [])
+	{
+		$this->_log(self::METHOD_UNSUPPORTED_BY_REDIS_BACKEND);
+		return array();
```
Comment (vaionicle) [inline library/Zend/Cache/Backend/Rediscluster.php]: []

File: library/Zend/Cache/Backend/Rediscluster.php [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+			$tags = [$tags];
+
+		$return = [];
+		foreach ($tags as $tag) {
+			foreach ($this->_redis->sMembers($this->_keyFromTag($tag)) as $id) {
+				$return[] = $id;
+			}
+		}
+		return $return;
+	}
```
Comment (vaionicle) [inline library/Zend/Cache/Backend/Rediscluster.php]: if conditions without { }

File: tests/Zend/Cache/RedisclusterBackendTest.php [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+		$this->_instance = new Zend_Cache_Backend_Rediscluster($options);
+		parent::setUp($notag);
+	}
+
+	public function tearDown()
+	{
+		parent::tearDown();
+		$this->_instance = null;
+		// We have to wait after a memcached flush
+//		sleep(1);
```
Comment (vaionicle) [inline tests/Zend/Cache/RedisclusterBackendTest.php]: remove comments

File: tests/Zend/Cache/RedisclusterBackendTest.php [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+/**
+ * Common tests for backends
+ */
+require_once 'CommonExtendedBackendTest.php';
+
+class Zend_Cache_Backend_RedisclusterBackendTest extends Zend_Cache_CommonExtendedBackendTest
+{
+	/**
+	 *
+	 * @var TP24_Cache_Backend_Rediscluster
```
Comment (vaionicle) [inline tests/Zend/Cache/RedisclusterBackendTest.php]: TP24_Cache_Backend_Rediscluster ??

Comment (dbaltas) [reply tests/Zend/Cache/RedisclusterBackendTest.php]: 👍

File: library/Zend/Cache/Backend/Rediscluster.php [hunk 6]
```diff
@@ -0,0 +1,536 @@
...
+
+	/**
+	 * Return the value if is numeric or false if not
+	 *
+	 *
+	 * @param string $id
+	 * @return integer or boolean
```
Comment (ngiallelis) [inline library/Zend/Cache/Backend/Rediscluster.php]: Hey @vaionicle you missed this one in your review: `@return integer|false`

Comment (vaionicle) [reply library/Zend/Cache/Backend/Rediscluster.php]: Sorry @ngiallelis my man, I left something back for you, just to remind you the goodDays

**Review by vaionicle — APPROVED** [review]

**Pull request merged**